File: Daily Test/Python/Boxes & Gold Coins 13.1.2026.py
n=int(input())
a=list(map(int,input().split()))
days=0

while True:
    nxt=[a[0]]
    removed=False
    for i in range(1,len(a)):
        if a[i]>a[i-1]:
            removed=True
        else:
            nxt.append(a[i])
    if not removed:
        break
    days+=1
    a=nxt

print(days)

File: Daily Test/C++/Boxes & Gold Coins 13.1.2026.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n; cin>>n;
    vector<int> a(n);
    for(int&i:a) cin>>i;

    int days=0;
    while(true){
        vector<int> nxt;
        nxt.push_back(a[0]);
        bool removed=false;

        for(int i=1;i<n;i++){
            if(a[i]>a[i-1]) removed=true;
            else nxt.push_back(a[i]);
        }

        if(!removed) break;
        days++;
        a=nxt;
        n=a.size();
    }
    cout<<days;
}